gpytorch/lazy/zero_lazy_tensor.py

```diff
@@ -43,7 +43,14 @@ def _matmul(self, rhs):
         rhs_size_ind = -2 if rhs.ndimension() > 1 else -1
         if self.size(-1) != rhs.size(rhs_size_ind):
             raise RuntimeError("Size mismatch, self: {}, rhs: {}".format(self.size(), rhs.size()))
-        return rhs * 0
+        new_m = self.size(-2)
+        if rhs_size_ind == -1:
+            *batch_shape, m = rhs.shape
+            output_shape = (*batch_shape, new_m)
+        else:
+            *batch_shape, m, n = rhs.shape
+            output_shape = (*batch_shape, new_m, n)
+        return torch.zeros(*output_shape)
```

Comment:
```suggestion
        return torch.zeros(*output_shape, device=rhs.device, dtype=rhs.dtype)
```
At a minimum, these should inherit